File: 프로그래머스/0/120875. 평행/평행.py
def solution(dots):
    check = [(0,1), (0,2), (0,3), (1,2), (1,3), (2,3)]
    count = 0
    for ch in check:
        i, j = ch
        num = [0, 1, 2, 3]
        num.remove(i)
        num.remove(j)
        k, l = num
        inc_1 = (dots[j][1]-dots[i][1])/(dots[j][0]-dots[i][0])
        inc_2 = (dots[l][1]-dots[k][1])/(dots[l][0]-dots[k][0])
        if inc_1 == inc_2:
            return 1
    return 0

File: 프로그래머스/0/120875. 평행/평행.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> dots) {
    int answer = 0;
    vector<vector<int>> check = {{0, 1}, {0, 2}, {0, 3}, {1, 2}, {1, 3}, {2, 3}};
    for(vector<int> v : check){
        vector<int> nums = {0, 1, 2, 3};
        int i = v[0];
        int j = v[1];
        nums.erase(remove(nums.begin(), nums.end(), i), nums.end());
        nums.erase(remove(nums.begin(), nums.end(), j), nums.end());
        int k = nums[0];
        int l = nums[1];
        double inc1 = (double)(dots[i][1]-dots[j][1])/(double)(dots[i][0]-dots[j][0]);
        double inc2 = (double)(dots[k][1]-dots[l][1])/(double)(dots[k][0]-dots[l][0]);
        if (inc1 == inc2) answer = 1;
    }
    return answer;
}